lib-multisrc/madara/src/eu/kanade/tachiyomi/multisrc/madara/Madara.kt

```diff
@@ -912,16 +1063,16 @@ abstract class Madara(
     /**
      * Fetch the genres from the source to be used in the filters.
      */
-    protected open fun fetchGenres() {
-        if (fetchGenres && fetchGenresAttempts <= 3 && (genresList.isEmpty() || fetchGenresFailed)) {
-            val genres = runCatching {
-                client.newCall(genresRequest()).execute()
+    protected fun fetchGenres() {
+        if (fetchGenres && fetchGenresAttempts < 3 && (genresList.isEmpty() || fetchGenresFailed)) {
+            try {
+                genresList = client.newCall(genresRequest()).execute()
                     .use { parseGenres(it.asJsoup()) }
+            } catch (_: Exception) {
+                fetchGenresFailed = true
+            } finally {
+                fetchGenresAttempts++
             }
-
-            fetchGenresFailed = genres.isFailure
-            genresList = genres.getOrNull().orEmpty()
-            fetchGenresAttempts++
         }
     }
```

Comment:
Should be doable without `fetchGenresFailed`? The variable isn't used outside of here.
```kt
protected fun fetchGenres() {
    if (!fetchGenres || fetchGenresAttempts >= 3 || genresList.isNotEmpty()) return

    try {
        genresList = client.newCall(genresRequest()).execute()
            .use { parseGenres(it.asJsoup()) }
    } catch (_: Exception) {
    }

    fetchGenresAttempts++
}
```